src/main/java/com/karankumar/bookproject/ui/shelf/CustomShelfForm.java

```diff
@@ -182,6 +240,7 @@ public CustomShelf getCustomShelf() {
     public static class DeleteEvent extends CustomShelfForm.CustomShelfFormEvent {
         DeleteEvent(CustomShelfForm source, CustomShelf customShelf) {
             super(source, customShelf);
+            source.customShelfService.delete(customShelf);
```

Comment:
May be we can add a test for this as well?

Reply:
Would the test built for checking the delete button's functionality cover this as well?

Reply:
I personally would write a test now instead of leaving it for later.